### PR TITLE
Allow specifying a channel for the system rustc

### PR DESCRIPTION
**Pull request by jflatow**
This is useful if you want to use e.g. `{:system, "+nightly"}`

### CODE REVIEW

File: rustler_mix/lib/rustler/compiler.ex [hunk 1]
```diff
@@ -43,8 +43,8 @@ defmodule Rustler.Compiler do
   end
 
   defp make_base_command(:system), do: ["cargo", "rustc"]
+  defp make_base_command({:system, channel}), do: ["cargo", channel, "rustc"]
```
Comment (evnu) [inline rustler_mix/lib/rustler/compiler.ex:46]: This should be documented in [rustler.ex](https://github.com/rusterlium/rustler/blob/master/rustler_mix/lib/rustler.ex#L22)

Comment (jflatow) [reply rustler_mix/lib/rustler/compiler.ex:49]: Added

**Comment by evnu** [comment]
The format `"+nightly"` might come as a surprise to users: If they use something like `{:system, "nightly"}`, `cargo` will think that the channel is actually an unknown command. We should either document the expected format with `+` or use atoms and translate (`{:system, :nightly}`). What do you think? Documenting the required `+` is probably the simplest thing to do here.

**Comment by jflatow** [comment]
Added some clarification, let me know what you think.

**Comment by evnu** [comment]
I like it, thanks!

**Comment by evnu** [comment]
@jflatow Can you fix the format error?

**Comment by jflatow** [comment]
Should be passing CI now, though I don't see the checks running at all again

**Comment by evnu** [comment]
> Should be passing CI now, though I don't see the checks running at all again

Seems that I need to run that explicitly. CI went through, thank you!

**Pull request merged**